lib/data/models/user/queries/get_authenticated_user.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:mm_flutter_app/data/models/base/base_operation.dart';
+
+class GetAuthenticatedUser implements BaseOperation {
```

Comment:
We might not want these to map 1:1 from dart queries to backend queries.

Some places, we may want to get a lot more data or different fields. For example, if I'm editing my own profile, most of the fields directly on the query will need to be returned. Several additional fields (countryOfResidence, ethnicity, firstName, lastName, birthYear, cityOfResidence, spokenLanguages, websites, etc.) will need to be returned here which generally won't be needed in other places where the `GetAuthenticatedUser` query is made.

Would we plan on adding a new `class GetAuthenticatedUserFullProfile` or some similar logic for the "edit profile" functionality? Or does this negate the benefits of standardizing these queries across the app, and we should just add all of the necessary fields to one `GetAuthenticatedUser` class?

Reply:
Yes I was thinking about this as well and I agree that we could create additional classes for different versions of the same query. However, I imagine we would only do this for when there is a huge difference in the amount of data returned between the versions, as was your example.